File: Week8Prolog/fizzbuzz.pl
fizzbuzz(X) :-
    0 is X mod 15,
    format('Fizzbuzz~n').

fizzbuzz(X) :-
    0 is X mod 3,
    format('Fizz~n').

fizzbuzz(X) :-
    0 is X mod 5,
    format('Buzz~n').

fizzbuzz(X) :-
    format('~w~n', [X]).

run_fizzbuzz(X, Y) :-
    X =< Y,
    fizzbuzz(X), !,
    Z is X + 1,
    run_fizzbuzz(Z, Y).

run_fizzbuzz(_,_) :- !.

:- run_fizzbuzz(1, 100).

File: Week3Pascal/fizzbuzz.pas
program fizzbuzz;

var 
    i: integer;

begin
  
  for i := 1 to 100 do 
    begin
        if i mod 15 = 0 then 
            writeln('Fizzbuzz')
        else if i mod 3 = 0 then
            writeln('Fizz')
        else if i mod 5 = 0 then
            writeln('Buzz')
        else 
            writeln(i);
    end;
  
end.
